### DISCUSSION TITLE
isValid returns false if I use with yup validations

### DISCUSSION THREAD

**Discussion by tMukodaGIG** [react-hook-form/react-hook-form#3099] [Q&A]
Hello! 

In the following code, I use  
```<input type="submit" disabled={!isValid} />``` 
to toggle the validity of submit button. However, isValid keeps returning false, even though displayed input fields are correctly filled in. This happens when I make validations with yup. Could you help me to solve this problem?

Please check this codesandbox.
[https://codesandbox.io/s/react-hook-form-validationschema-v6-forked-9ezus](https://codesandbox.io/s/react-hook-form-validationschema-v6-forked-9ezus)

~~~
import React from "react";
import ReactDOM from "react-dom";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";

import "./styles.css";

const SignupSchema = yup.object().shape({
  firstName: yup.string().required(),
  lastName: yup.string(),
  age: yup.number().required().positive().integer(),
  website: yup.string().url()
});

function App() {
  const { register, handleSubmit, errors, formState } = useForm({
    mode: "onChange",
    resolver: yupResolver(SignupSchema)
  });
  const [isShow, setIsShow] = React.useState(false);

  const onSubmit = (data) => {
    alert(JSON.stringify(data));
  };

  const { isValid } = formState;

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {isShow ? (
        <>
          <div>
            <label>First Name</label>
            <input type="text" name="firstName" ref={register} />
            {errors.firstName && <p>{errors.firstName.message}</p>}
          </div>
          <div style={{ marginBottom: 10 }}>
            <label>Last Name</label>
            <input type="text" name="lastName" ref={register} />
            {errors.lastName && <p>{errors.lastName.message}</p>}
          </div>
        </>
      ) : (
        <>
          <div>
            <label>Age</label>
            <input type="text" name="age" ref={register} />
            {errors.age && <p>{errors.age.message}</p>}
          </div>
          <div>
            <label>Website</label>
            <input type="text" name="website" ref={register} />
            {errors.website && <p>{errors.website.message}</p>}
          </div>
        </>
      )}
      <button onClick={() => setIsShow((p) => !p)}>toggle</button>
      <input type="submit" disabled={!isValid} />
    </form>
  );
}

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);

~~~

**Comment by bluebill1049** [comment]
what's the issue and what's the steps to reproduce the problem. CSB link is not your example as well.

**Comment by tMukodaGIG** [comment]
@bluebill1049 
I am sorry for my vague explanation. I changed the example as well. 
Please check this codesandbox: [https://codesandbox.io/s/yupisvalidtest-9ezus](https://codesandbox.io/s/yupisvalidtest-9ezus)

### About the form
・If you click on the "show input for age" check button, an input field for Age shows up. 
・I use Yup to set both Name and Age required
・I use isValid to toggle the validity of submit button
![スクリーンショット 2020-10-07 10 13 24](https://user-images.githubusercontent.com/62982913/95276294-bf6a0100-0885-11eb-99ce-bdcce2497613.png)


### Problem
Submit button is disabled even though the Age input field is not rendered. Is it possible to make the button enabled when Name field is filled and Age field is not rendered? What I want to do is that if a user decides to submit their name, only Name field is required. If they decide to submit both their name and age, both Name and Age fields are required. 
![スクリーンショット 2020-10-07 10 06 20](https://user-images.githubusercontent.com/62982913/95276474-40c19380-0886-11eb-95b7-333c5bfce96e.png)

Please let me know if you need more details. Thank you very much for your help.

**Reply by bluebill1049** [reply]
what about if you just toggle the visibility of that input? so it's actually always mounted.

**Reply by tMukodaGIG** [reply]
@bluebill1049 
Thank you for the reply. 
I used "visibility" to toggle the visibility of the Age input. However, the form is still invalid when the Name input is filled and the Age input is invisible...
[https://codesandbox.io/s/yupisvalidtest-9ezus?file=/src/index.js](https://codesandbox.io/s/yupisvalidtest-9ezus?file=/src/index.js)

![スクリーンショット 2020-10-07 10 34 43](https://user-images.githubusercontent.com/62982913/95277580-191ffa80-0889-11eb-9094-1e5e42e73719.png)

**Reply by bluebill1049** [reply]
<img width="1190" alt="Screen Shot 2020-10-07 at 12 58 53 pm" src="https://user-images.githubusercontent.com/10513364/95278842-db2cd180-089c-11eb-9871-f688193d7882.png">

it's working as expected for me.

**Reply by tMukodaGIG** [reply]
@bluebill1049 
I am sorry for the confusion. I would like to make the form submittable when the Name field is filled without showing the Age field. 
![スクリーンショット 2020-10-07 11 04 54](https://user-images.githubusercontent.com/62982913/95279264-3bb41280-088d-11eb-9dd8-380d0cff3614.png)

**Comment by bluebill1049** [comment]
Then ur age input need to be valid value.

Cheers
Bill

On 7 Oct 2020, at 1:07 pm, TKC <notifications@github.com<mailto:notifications@github.com>> wrote:


@bluebill1049<https://github.com/bluebill1049>
I am sorry for the confusion. I would like to make the form submittable when the Name field is filled without showing the Age field.
[スクリーンショット 2020-10-07 11 04 54]<https://user-images.githubusercontent.com/62982913/95279264-3bb41280-088d-11eb-9dd8-380d0cff3614.png>

**Reply by tMukodaGIG** [reply]
@bluebill1049 
OK.. so, it is impossible to ignore the validations for Age when the filed is invisible or not rendered?

**Comment by DarkSouls101-zz** [comment]
U will probably have to build ur own custom resolver (with context) to ignore those input which r hidden. 

Cheers
Bill

**Reply by mkelley33** [reply]
@bluebill1049 I'm getting the same results for isValid.  I tried it your way but isValid always comes back false.  See the discussion and my implementation:

https://github.com/orgs/react-hook-form/discussions/10653

I may try something besides yup to see if the yupResolver is the problem as I suspect.
Note, I'm also running this in a gatsby project, if that makes any difference.
